Modules/_ctypes/callproc.c

```diff
@@ -1014,7 +1014,10 @@ void _ctypes_extend_error(PyObject *exc_class, const char *fmt, ...)
 
     PyErr_Fetch(&tp, &v, &tb);
     PyErr_NormalizeException(&tp, &v, &tb);
-    cls_str = PyObject_Str(tp);
+    if (PyType_Check(tp))
+        cls_str = PyUnicode_FromString(_PyType_Name((PyTypeObject *)tp));
```

Comment:
Confirmed that we don't have to guard against `_PyType_Name` returning NULL: https://github.com/python/cpython/blob/14c13955c59c15bd58520241ce64e27c68028d64/Objects/typeobject.c#L489